### PR TITLE
Overhaul tests in urdf and kdl_parser

### PR DESCRIPTION
**Pull request by jackieokay**
the rostests in urdf and kdl_parser were failing. To make smaller, isolated test cases, I decided to change the urdf test in both packages. I also overhauled how the urdf tests work,

The tests did not pass locally for me unless I built the packages with a source build of urdfdom. I believe this has to do with the fact that the Trusty deb was built without ros/urdfdom#71

An outstanding question is why nobody has ever noticed these failing tests before. Shouldn't the build farm have notified someone about them?

### CODE REVIEW

**Comment by scpeters** [comment]
This isn't built by the ros build farm. We should enable travis for all these packages.

**Comment by jackieokay** [comment]
Do you know how to enable Travis on the ROS build farm? Because I have no idea.

**Comment by scpeters** [comment]
from [ros-users](http://lists.ros.org/pipermail/ros-users/2015-October/069723.html):
- https://github.com/AutonomyLab/bebop_autonomy/blob/indigo-devel/.travis.yml

**Comment by scpeters** [comment]
Travis support added in #121

**Comment by scpeters** [comment]
Can you rebase and we'll see what travis says?

File: kdl_parser/test/test_robot.urdf [hunk 1]
```diff
@@ -0,0 +1,425 @@
+<?xml version="1.0"?>
+<robot name="r2d2">
+
+<material name="blue">
```
Comment (scpeters) [inline kdl_parser/test/test_robot.urdf]: indentation confused me here. Mind if add two spaces to this line?

Comment (scpeters) [inline kdl_parser/test/test_robot.urdf]: fixed in d586847d6e747dffe5c8fa774ade19372db03c88

**Comment by scpeters** [comment]
+1

**Pull request merged**